### PR TITLE
Enhance USER_DISABLED_FEATURES to allow disabling change username or full name

### LINKED ISSUE #31958: Option to lock full name when using external auth sources
### Feature Description

In certain scenarios (such as within schools), users registered through external authentication need to maintain their real names. 

Since names may contain Unicode characters (like a Chinese name), a good practice is to set `DEFAULT_SHOW_FULL_NAME=true` and set the user's full name to their real name. In this situation, it's necessary to ensure that external users cannot modify their own full name.

An option like `EXTERNAL_USER_LOCK_FULL_NAME` would be great to fit this need.

### Screenshots

image

### PR DESCRIPTION
**Pull request by thezzisu**
Fix #31958
Enhanced `USER_DISABLED_FEATURES`(also `EXTERNAL_USER_DISABLE_FEATURES`) option in `[admin]` section.
Added following values:
- `change_username`: Disable change username
- `change_full_name`: Disable change full name
---

Progress:
- [x] Update code
- [x] Update translations

### CODE REVIEW

**Comment by lunny** [comment]
Please don't submit translations except en-US. We use https://translate.gitea.com to do that work.

**Comment by thezzisu** [comment]
> Please don't submit translations except en-US. We use https://translate.gitea.com to do that work.

Thank you! Translations re-updated in [7128278](https://github.com/go-gitea/gitea/commit/71282780a35f02989986dcfe74034acc0255973f).

**Comment by techknowlogick** [comment]
Thank you for this PR, are you able to combine this into `USER_DISABLED_FEATURES`/`EXTERNAL_USER_DISABLE_FEATURES`? Or if that doesn't make sense, then a more genric option that accepts a list of restricted items from being changed?

**Comment by thezzisu** [comment]
> Thank you for this PR, are you able to combine this into `USER_DISABLED_FEATURES`/`EXTERNAL_USER_DISABLE_FEATURES`? Or if that doesn't make sense, then a more genric option that accepts a list of restricted items from being changed?

Thank you for your advice! I have updated my implementation and now its using `USER_DISABLED_FEATURES` to work.

**Comment by thezzisu** [comment]
Action `db-tests / test-mssql` seems to be broken.

**Comment by lunny** [comment]
> Action `db-tests / test-mssql` seems to be broken.

This has been resolved.

**Comment by lunny** [comment]
app.example.ini should be updated and a documentation PR should be created in https://gitea.com/gitea/docs

**Comment by thezzisu** [comment]
> USER_DISABLED_FEATURES

Thank you! `app.example.ini` is updated and related docs is in a PR via https://gitea.com/gitea/docs/pulls/76

File: templates/user/settings/profile.tmpl [hunk 1]
```diff
@@ -12,14 +12,17 @@
 						<span class="text red tw-hidden" id="name-change-prompt"> {{ctx.Locale.Tr "settings.change_username_prompt"}}</span>
 						<span class="text red tw-hidden" id="name-change-redirect-prompt"> {{ctx.Locale.Tr "settings.change_username_redirect_prompt"}}</span>
 					</label>
-					<input id="username" name="name" value="{{.SignedUser.Name}}" data-name="{{.SignedUser.Name}}" autofocus required {{if or (not .SignedUser.IsLocal) .IsReverseProxy}}disabled{{end}} maxlength="40">
-					{{if or (not .SignedUser.IsLocal) .IsReverseProxy}}
+					<input id="username" name="name" value="{{.SignedUser.Name}}" data-name="{{.SignedUser.Name}}" autofocus required {{if or (or (not .SignedUser.IsLocal) ($.UserDisabledFeatures.Contains "change_username")) .IsReverseProxy}}disabled{{end}} maxlength="40">
+					{{if or (or (not .SignedUser.IsLocal) ($.UserDisabledFeatures.Contains "change_username")) .IsReverseProxy}}
```
Comment (lunny) [inline templates/user/settings/profile.tmpl]: I think `or` can accept 2, 3 or more parameters, so the code can be simplified.

Comment (thezzisu) [reply templates/user/settings/profile.tmpl]: That's surely a better approach. Code simplified in latest commit.

**Comment by manishramanandi** [comment]
hey i want to contribute can you explain me a lil bit more??

**Comment by lunny** [comment]
This is the backport pull request to resolve the problem, not the issue.

File: templates/user/settings/profile.tmpl [hunk 2]
```diff
@@ -12,14 +12,17 @@
 						<span class="text red tw-hidden" id="name-change-prompt"> {{ctx.Locale.Tr "settings.change_username_prompt"}}</span>
 						<span class="text red tw-hidden" id="name-change-redirect-prompt"> {{ctx.Locale.Tr "settings.change_username_redirect_prompt"}}</span>
 					</label>
-					<input id="username" name="name" value="{{.SignedUser.Name}}" data-name="{{.SignedUser.Name}}" autofocus required {{if or (not .SignedUser.IsLocal) .IsReverseProxy}}disabled{{end}} maxlength="40">
-					{{if or (not .SignedUser.IsLocal) .IsReverseProxy}}
+					<input id="username" name="name" value="{{.SignedUser.Name}}" data-name="{{.SignedUser.Name}}" autofocus required {{if or (or (not .SignedUser.IsLocal) ($.UserDisabledFeatures.Contains "change_username")) .IsReverseProxy}}disabled{{end}} maxlength="40">
```
Comment (lunny) [inline templates/user/settings/profile.tmpl]: The same for `or` usage.

Comment (thezzisu) [reply templates/user/settings/profile.tmpl]: Sorry for this mistake. Fixed in next commit

**Review by techknowlogick — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**